### PR TITLE
android: Fix search not filtering machines from input

### PR DESCRIPTION
**Pull request by joshvocal**
Resolves: https://github.com/tailscale/tailscale/issues/13218

* Typing into the textfield to filter machines properly works now
  * `searchTerm.collect {}` was unreachable and caused it not to collect input

| Before | After |
|--------|--------|
| <video src="https://github.com/user-attachments/assets/20eea88c-0458-409c-b114-17517bb1c9d7" /> | <video src="https://github.com/user-attachments/assets/c74c8f1c-6454-4247-851a-6869719d84c6" /> |

### CODE REVIEW

File: android/src/main/java/com/tailscale/ipn/ui/viewModel/MainViewModel.kt [hunk 1]
```diff
@@ -132,10 +143,6 @@ class MainViewModel(private val vpnViewModel: VpnViewModel) : IpnViewModel() {
         }
       }
 
-      viewModelScope.launch {
-        searchTerm.collect { term -> peers.set(peerCategorizer.groupedAndFilteredPeers(term)) }
-      }
-
       viewModelScope.launch {
```
Comment (kari-ts) [inline android/src/main/java/com/tailscale/ipn/ui/viewModel/MainViewModel.kt:146]: I believe the bug was introduced when a } was deleted and this block and the above block launching the searchTerm.collect became unreachable. Could you also move the `viewModelScope.launch {
      App.get().healthNotifier?.currentIcon?.collect { icon -> healthIcon.set(icon) }
    }` as well please?

**Review by kari-ts — CHANGES_REQUESTED** [review]
Thanks for finding the bug and creating a fix for it! We really appreciate it!

File: android/src/main/java/com/tailscale/ipn/ui/viewModel/MainViewModel.kt (hunk 1 above)
Comment (joshvocal) [reply android/src/main/java/com/tailscale/ipn/ui/viewModel/MainViewModel.kt]: I've made those changes @kari-ts

**Review by kari-ts — APPROVED** [review]
Thanks for the fix!

**Pull request merged**